Makefile

```diff
@@ -702,51 +708,52 @@ localmachine:
 	@echo /define.gitCommit=$(GIT_COMMIT)
 	$(MAKE) ginkgo-run GINKGO_PARALLEL=n TAGS="$(REMOTETAGS)" GINKGO_FLAKE_ATTEMPTS=0 FOCUS_FILE=$(FOCUS_FILE) GINKGOWHAT=pkg/machine/e2e/.
 
+REMOTESYSTEM_TCP_PORT ?= 8080
+REMOTESYSTEM_TLS_CA_CRT ?= $(CURDIR)/bin/remotesystem.ca.crt.pem
+REMOTESYSTEM_TLS_CA_KEY ?= $(CURDIR)/bin/remotesystem.ca.key.pem
+REMOTESYSTEM_TLS_SERVER_CRT ?= $(CURDIR)/bin/remotesystem.server.crt.pem
+REMOTESYSTEM_TLS_SERVER_KEY ?= $(CURDIR)/bin/remotesystem.server.key.pem
+REMOTESYSTEM_TLS_CLIENT_CRT ?= $(CURDIR)/bin/remotesystem.client.crt.pem
+REMOTESYSTEM_TLS_CLIENT_KEY ?= $(CURDIR)/bin/remotesystem.client.key.pem
+REMOTESYSTEM_TLS_BOGUS_CRT ?= $(CURDIR)/bin/remotesystem.bogus.crt.pem
+REMOTESYSTEM_TLS_BOGUS_KEY ?= $(CURDIR)/bin/remotesystem.bogus.key.pem
+
+export \
+	REMOTESYSTEM_TCP_PORT \
+	REMOTESYSTEM_TLS_CA_KEY \
+	REMOTESYSTEM_TLS_CA_CRT \
+	REMOTESYSTEM_TLS_SERVER_CRT \
+	REMOTESYSTEM_TLS_SERVER_KEY \
+	REMOTESYSTEM_TLS_CLIENT_CRT \
+	REMOTESYSTEM_TLS_CLIENT_KEY \
+	REMOTESYSTEM_TLS_BOGUS_CRT \
+	REMOTESYSTEM_TLS_BOGUS_KEY
+
+$(REMOTESYSTEM_TLS_CA_CRT) $(REMOTESYSTEM_TLS_CA_KEY) \
+$(REMOTESYSTEM_TLS_CLIENT_CRT) $(REMOTESYSTEM_TLS_CLIENT_KEY) \
+$(REMOTESYSTEM_TLS_SERVER_CRT) $(REMOTESYSTEM_TLS_SERVER_KEY) \
+$(REMOTESYSTEM_TLS_BOGUS_CRT) $(REMOTESYSTEM_TLS_BOGUS_KEY) \
+&:
+	source hack/remotesystem.env ; remotesystem-gen-tls
```

Comment:
IMO this has nothing to do in the Makefile, it bloats the Makefile with a bunch of unrelated options.

More importantly it creates a dependency on the makefile targets to run the bast tests which is simply wrong.

The bats tests must be able to be run on their own via just `bats test/system`, in fact we just ship the test/system directory in the test/system rpm. to run them as part of distro level gating on reverse dependency testing.
There is also the the way to run hack/bats for local dev testing.

So if we need additional setup such as creating temporary certs it must always be encapsulated in the bats setup themselves.

Reply:
I can move the actual cert generation to the bats setup helpers, but these variables must be known to the makefile, as the makefile previously and still does, now through hack/remotesystem.env, perform a check to ensure there is no podman service already running as well as to wait for the newly launched service to start.